qiskit_experiments/calibration_management/base_calibration_experiment.py

```diff
@@ -373,6 +379,16 @@ def _add_cal_metadata(self, circuits: List[QuantumCircuit]):
         """
         pass
 
+    def _circuit_instructions(self) -> List[str]:
+        """Return a list of circuit instructions used."""
+
+        instruction_names = set()
+        for circuit in self.circuits():
+            for inst in circuit.data:
+                instruction_names.add(inst[0].name)
```

Comment:
For multi-qubit circuits, this will potentially have some false positives.

Reply:
Perhaps, do you have an example in mind?

Reply:
Well, this function by itself is fine. It will return all of the instruction names. But how it is used below to query the parameters_table it will query for single qubit gates like `x` and `sx` for both qubits even if only one of the qubits uses an `x` or `sx` gate in the circuits. It might be too much trouble to try to exclude these gates.

Reply:
This is outdated with the most recent commits.